packages/@storefront/flux-capacitor/test/unit/core/emitter.ts

```diff
@@ -0,0 +1,293 @@
+        'Hello, world!',
+      ];
+
+      vals.forEach((val) => expect(() => emitter.all(val , () => null)).to.throw(error));
+    });
+
+    it('should ignore empty arrays', () => {
+      emitter.all([], () => null);
+
+      expect(emitter._barriers).to.eql({});
+      expect(emitter._lookups).to.eql({});
+    });
+
+    it('should set the _barriers and _lookups objects with the events and callback', () => {
+      emitter.all(events1, callback1);
+
+      expect(emitter._lookups[events1[0]]).to.eql([key1]);
+      expect(emitter._lookups[events1[1]]).to.eql([key1]);
+      expect(emitter._barriers[key1]).to.eql({
+        events: { a: 0, b: 0 },
+        callbacks: [{ callback: callback1, context: emitter }],
+      });
+    });
+
+    it('should allow the context to be specified' , () => {
+      const that = { c: 'd' };
+
+      emitter.all(events1, callback1, that);
+
+      expect(emitter._barriers[key1]).to.eql({
+        events: { a: 0, b: 0 },
+        callbacks: [{ callback: callback1, context: that }],
+      });
+    });
+
+    it('should update the callbacks array when an event already exists in _barriers', () => {
+      emitter.all(events1, callback1);
+      emitter.all(events1, callback2);
+
+      expect(emitter._barriers[key1]).to.eql({
+        events: { a: 0, b: 0 },
+        callbacks: [
+          { callback: callback1, context: emitter },
+          { callback: callback2, context: emitter },
+        ],
+      });
+    });
+
+    it('should sort the events before generating the key', () => {
+      emitter.all(events1, callback1);
+      emitter.all(events1.reverse(), callback2);
+
+      expect(emitter._barriers[key1].callbacks).to.have.length(2);
+    });
+
+    it('should update the _lookups object with a new key', () => {
+      emitter.all(events1, callback1);
+      emitter.all(events2, callback2);
+
+      expect(emitter._lookups).to.eql({ a: [key1, key2], b: [key1], c: [key2] });
+    });
+
+    it('should not register duplicate keys', () => {
+      emitter._lookups = { a: [key1], b: [key1] };
+      emitter._barriers = {
+        [key1]: {
+          events: { a: 0, b: 0 },
+          callbacks: [{ callback: callback1, context: emitter }],
+        },
+      };
+
+      emitter.all(events1, callback2);
+
+      expect(emitter._lookups[events1[0]]).to.eql([key1]);
+      expect(emitter._lookups[events1[1]]).to.eql([key1]);
+    });
+
+    it('should not reset the event counters', () => {
+      emitter._lookups = { a: [key1], b: [key1] };
+      emitter._barriers = {
+        [key1]: {
+          events: { a: 0, b: 420 },
+          callbacks: [{ callback: callback1, context: emitter }],
+        },
+      };
+
+      emitter.all(events1, callback2);
+
+      expect(emitter._barriers[key1].events).to.eql({ a: 0, b: 420 });
+    });
+  });
+
+  describe('allOff', () => {
+    let events1;
+    let events2;
+    let key1;
+    let key2;
+    let callback1;
+    let callback2;
+
+    before(() => {
```

Comment:
The tests are not isolated with this as a `before`. This should be a `beforeEach`.

Reply:
Implemented.